### DISCUSSION TITLE
why the menu is outside of the frame?

### DISCUSSION THREAD

**Discussion by geextahslex** [christoph-heinrich/mpv-quality-menu#40] [Q&A] [answered]
Hi, I'm trying this script out at the moment and when I press `F` which is bound in the `input.conf` like in the description, the menu shows up outside of the mpv frame.
mpv: v0.38.0

Any ideas?

Thank you :)
![menu broken](https://github.com/christoph-heinrich/mpv-quality-menu/assets/139760266/e5a25e74-1c29-4aad-854a-24746fa846f8)

**Comment by geextahslex** [comment]
okay after playing around with
```
# paddings for top left corner
text_padding_x=500
text_padding_y=500
```
no you can see the list but it still looks messed up
Is it possible to hide resolutions below 1080p?
![asdda](https://github.com/christoph-heinrich/mpv-quality-menu/assets/139760266/b6b0db36-03a5-40f4-83f3-f3ff3dc3c1db)

**Reply by christoph-heinrich** [reply]
> Is it possible to hide resolutions below 1080p?

No, but there is an open feature request for that https://github.com/christoph-heinrich/mpv-quality-menu/issues/36.
If you have any ideas on what such an option should look like, then feel free to leave a suggestion :wink:

**Reply by geextahslex** [reply]
Yes that sounds like a nice quality of life improvement. I have no idea about programing so I don't know how much helpfull input I can provide.
So if he draws the line and knows about resolution/codec/filesize and FPS there could be a if then condition, and you could put in your threshold. For example show only `resolution<1440p` `codec=VP9` and `filesize <1mb` if all conditions are met, then show line

**Comment by christoph-heinrich** [comment, marked as answer]
Looks to me like you're using some sort of style overwrite in your `quality-menu.conf` that centers the output like `\an5`.
Most likely the option `style_ass_tags`, but it could also be achieved with the 4 prefix options (`selected_and_active` and similar).

It might also be possible that you have something in your `mpv.conf` that's at fault, but the latest version version of this script should reset everything to the default before drawing.

If you can't find the problem, you can try changing [this line](https://github.com/christoph-heinrich/mpv-quality-menu/blob/96e449dc6b2eb0a9986bcb5d55c07a21f1a1d36e/quality-menu.lua#L751) to 
```lua
ass:append('{\\rDefault\\an7\\q2\\clip(' .. clipping_coordinates .. ')}' .. opts.style_ass_tags)
```

**Reply by geextahslex** [reply]
Hi, thank you for your answer

>Looks to me like you're using some sort of style overwrite in your quality-menu.conf

I just used the default file, so there were no changes to the file. 

>It might also be possible that you have something in your mpv.conf that's at fault, but the latest version version of this script should reset everything to the default before drawing.

I don't know. That is possible. I also used the default .lua file. So this was my first user experience, with the menu out of bounce ^^'

>If you can't find the problem, you can try changing this line to

ass:append('{\\rDefault\\an7\\q2\\clip(' .. clipping_coordinates .. ')}' .. opts.style_ass_tags)

That helped. Thank you for that. But I still need to use `scale_playlist_by_window=no` because otherwise it is simply too big ^^
Now it looks like this
![fixed](https://github.com/christoph-heinrich/mpv-quality-menu/assets/139760266/c6162013-37d1-40a7-a8b9-09fee55c16bd)

I have also a question about the 2 second timeout when you move in the menu. So with `menu_timeout=0` when I open the menu and don't move then it stays open but when I move up or down then it closes after 2 seconds. How can I disable that timeout?

**Reply by christoph-heinrich** [reply]
> I don't know. That is possible. I also used the default .lua file. So this was my first user experience, with the menu out of bounce ^^'

There has to be something in your config screwing it up. I'll simply add a commit with that `\an7` then it doesn't matter.

> But I still need to use `scale_playlist_by_window=no` because otherwise it is simply too big

What do you mean with too big? you can change the font size if the text is too big, and the menu can scroll if there isn't enough vertical space.


> So with `menu_timeout=0` when I open the menu and don't move then it stays open but when I move up or down then it closes after 2 seconds.

Can't reproduce. When I set the timeout to 0 the menu stays open indefinitely, doesn't matter if I move around by keeping the up/down key pressed, or by repeatedly hitting that key, or if I don't do anything.

Also the default timeout is 6 seconds, so I don't know where you're getting those 2 seconds from lol

**Reply by geextahslex** [reply]
> What do you mean with too big? you can change the font size if the text is too big, and the menu can scroll if there isn't enough vertical space.

I mean it looks like this
Is it possible to choose your own scale factor of the menu? Not only `scale_playlist_by_window=` `yes` or `no`  ? 
![scale yes](https://github.com/christoph-heinrich/mpv-quality-menu/assets/139760266/4c274316-81cc-4ae4-9ec2-d350b8961f47)

>Can't reproduce. When I set the timeout to 0 the menu stays open indefinitely, doesn't matter if I move around by keeping the up/down key pressed, or by repeatedly hitting that key, or if I don't do anything.
Also the default timeout is 6 seconds, so I don't know where you're getting those 2 seconds from lol

So after checking it has something to do with setting `menu_timeout=0` when I set it to 0 then it stays opened without moving in the list but closes after 1-2 seconds when you have moved. If I set it to a bigger value like 600 then it stays opened for that time no matter if you have moved or not. I don't know what causes this. It's actually okay this way, I mean I will not stare for 5 miutes at that list ^^

**Reply by christoph-heinrich** [reply]
> I mean it looks like this

I don't see a problem with that, if you move down the list will scroll.

> Is it possible to choose your own scale factor of the menu? Not only `scale_playlist_by_window=` `yes` or `no` ?

You can change the font size by changing the 25 in this option `style_ass_tags={\\fnmonospace\\fs25\\bord1}`

**Reply by geextahslex** [reply]
Thank you, awesome :)

**Comment by geextahslex** [comment]
Hi I have a different question. Do you know how to make a profile where the condition is the youtube uploaders name? And is it possible to switch the resolution depending on that? So for example if you watch youtuber xyz then the video is in 4k and if you watch xyz2 then the resolution is 1440p?

**Reply by christoph-heinrich** [reply]
That is not possible.
There is currently an [open PR for mpv](https://github.com/mpv-player/mpv/pull/14098) that might enable profile conditions based on uploader name, but there hasn't been any activity in a while, so no idea how long until that gets in.

**Reply by geextahslex** [reply]
Oh okay, thanks for that information :)
